adagrams/game.py

```diff
@@ -1,11 +1,131 @@
+
+import random
+from random import choices
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+POINTS = {
+    1:["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"],
+    2:["D","G"],
+    3:["B", "C", "M", "P"],
+    4:["F", "H", "V", "W", "Y" ],
+    5:["K"],
+    8:["J","X"],
+    10:["Q","Z"],
+}
+
+ADD_EIGHT_IF_LENGHT = [7, 8, 9, 10]
+
+
+def build_pile_of_letters(dictionary):
+    pile = []
+    if dictionary == []:
+        return None
+    for x,y in dictionary.items():
+        for i in range(0,y):
+            pile.append(x)
+    return pile
+
 def draw_letters():
-    pass
+    hand = random.sample(build_pile_of_letters(LETTER_POOL),10)
+    return hand
 
 def uses_available_letters(word, letter_bank):
-    pass
+    new_bank = list(letter_bank)
+    remainders = list(word.upper())
+    print(remainders)
+    
+    for i in word.upper(): 
+        if i in new_bank:
+            remainders.remove(i)
+            new_bank.remove(i)
+
+    if remainders == []:
+        return True
+    else: 
+        return False
+
 
 def score_word(word):
-    pass
+    points = 0
+    for i in word.upper():
+        for x,y in POINTS.items():
+            if i in y:
+                points += x
+    if len(word) in ADD_EIGHT_IF_LENGHT:
+        points += 8
+    
+    return points
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    pair_word_points= {}
+    max_score = 0
+    semifinalist = {}
+    winner = ()
+
+    for i in word_list:
+        pair_word_points[i]=score_word(i)
+
+    max_score= max(pair_word_points.values())
+
+    for i in pair_word_points:
+        if pair_word_points[i] == max_score:
+            semifinalist[i]=pair_word_points[i]
+
+    for x,y in semifinalist.items():
+        if len(x) == 10:
+            winner = (x,y)
+            return winner
+
+    if len(semifinalist) == 1:
+        winner = (list(semifinalist.keys())[0],list(semifinalist.values())[0])
+    else:
+        winner = (min(semifinalist.keys(),key=len),semifinalist[min(semifinalist.keys(),key=len)])
```

Comment:
This works, but it is very difficult to read. We are calling `min` twice for two different values. How might we dry this code up (aka "don't repeat yourself") or perhaps make it more verbose to help with readability?

It's a hard balance to meet when you want to take advantage of functions already built for us, shorter code lines, but also readability. 

Maybe we could combine the for loops on lines 98, 102, and 110 into one and iterate over `pair_word_points`, where we compare the score to our highest score and its length.  Then we can return a highest scored word and the highest score.